x-pack/plugins/security_solution/public/timelines/components/modal/header/index.tsx

```diff
@@ -0,0 +1,190 @@
+  const userCasesPermissions = cases.helpers.canUseCases([APP_ID]);
+
+  const title = useSelector((state: State) => selectTitleByTimelineById(state, timelineId));
+  const isDataInTimeline = useSelector((state: State) => selectDataInTimeline(state, timelineId));
+  const kqlQueryObj = useSelector((state: State) => selectKqlQuery(state, timelineId));
+
+  const { activeTab, dataProviders, timelineType, filters, kqlMode } = useSelector((state: State) =>
+    selectTimelineById(state, timelineId)
+  );
+
+  const combinedQueries = useMemo(
+    () =>
+      combineQueries({
+        config: esQueryConfig,
+        dataProviders,
+        indexPattern,
+        browserFields,
+        filters: filters ? filters : [],
+        kqlQuery: kqlQueryObj,
+        kqlMode,
+      }),
+    [browserFields, dataProviders, esQueryConfig, filters, indexPattern, kqlMode, kqlQueryObj]
+  );
+
+  const closeTimeline = useCallback(() => {
+    createHistoryEntry();
+    dispatch(timelineActions.showTimeline({ id: timelineId, show: false }));
+  }, [dispatch, timelineId]);
+
+  return (
+    <TimelinePanel
+      grow={false}
+      paddingSize="s"
+      hasShadow={false}
+      data-test-subj="timeline-modal-header-panel"
+    >
+      <EuiFlexGroup
+        className="eui-scrollBar"
+        alignItems="center"
+        gutterSize="s"
+        responsive={false}
+        justifyContent="spaceBetween"
+        css={autoOverflowXCSS}
+      >
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup gutterSize="xs" alignItems="center" responsive={false}>
+            <EuiFlexItem grow={false}>
+              <AddToFavoritesButton timelineId={timelineId} />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiText
+                grow={false}
+                data-test-subj="timeline-modal-header-title"
+                css={whiteSpaceNoWrapCSS}
+              >
+                <h3>{title}</h3>
+              </EuiText>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <TimelineSaveStatus timelineId={timelineId} />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup
+            id={TIMELINE_TOUR_CONFIG_ANCHORS.ACTION_MENU}
+            justifyContent="flexEnd"
+            alignItems="center"
+            gutterSize="xs"
+            responsive={false}
+            data-test-subj="timeline-modal-header-actions"
+          >
+            <EuiFlexItem>
+              <NewTimelineButton timelineId={timelineId} />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <OpenTimelineButton />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <InspectButton
+                queryId={`${timelineId}-${activeTab}`}
+                inputId={InputsModelId.timeline}
+                isDisabled={!isDataInTimeline || combinedQueries?.filterQuery === undefined}
```

Comment:
💅  nit: wdyt about moving this to a constant outside the JSX?

Reply:
I'm confused, which part could I extract in a constant?

Reply:
Sth like

```tsx
const isInspectDisabled = !isDataInTimeline || combinedQueries?.filterQuery === undefined;

return (
  (...)
    <InspectButton
      isDisabled={isInspectDisabled}
  (...)
)
```

Reply:
aaaaah yeah makes sense. I got confused when you said `outside the JSX` sorry!
Done!